src/main/scala/org/apache/spark/sql/redis/RedisSourceRelation.scala

```diff
@@ -142,8 +143,9 @@ class RedisSourceRelation(override val sqlContext: SQLContext,
           val conn = node.connect()
           foreachWithPipeline(conn, keys) { (pipeline, key) =>
             val row = rowsWithKey(key)
-            val encodedRow = persistence.encodeRow(keyName, row)
-            persistence.save(pipeline, key, encodedRow, ttl)
+            val encodedRow = persistence.encodeRow(keyName, row, ttlColumn)
+            val recordTTL = if (ttlColumn.isEmpty) ttl else row.getAs[Int](ttlColumn.get)
```

Comment:
You should handle the case where the user passes both ttl and ttl column somewhere.
Think a case statement would be nicer here

Reply:
have you got an update for this?

Reply:
Yes, I've followed their pattern to validate whether both values are set:
```
  if (ttlColumn.isDefined && ttl > 0) {
    throw new IllegalArgumentException(s"Both options '$SqlOptionTTL' and '$SqlOptionTTLColumn' are set. " +
      s"You should only use either one.")
  }
```

Reply:
Good stuff

Reply:
still think a case statement would be better than using ternary operators, but not a big issue